File: Leetle/JS_Solutions/90-checkifallelementsinalistarequnique.js
/*
Write a function solve that checks if all elements in a list are unique (no duplicates).

Example:
Input: [1, 2, 3, 4]
Output: True
*/

const solve = (arr) => {
    let set = new Set();
    for (let index = 0; index < arr.length; index++) {
        if (set.has(arr[index])) {
            return false
        }
        set.add(arr[index])
    }
    return true
}

console.log(solve([1,2,3,4]))
console.log(solve([1,2,1,4]))

// We can rewrite the for loop like this using the for ... of, so we don't have to handle index manually
const solve2 = (arr) => {
    let set = new Set();
    for (const num of arr) {
        if (set.has(num)) return false;
        set.add(num);
    }
    return true;
};

console.log(solve2([1,2,3,4]))
console.log(solve2([1,2,1,4]))

File: Leetle/90-checkifallelementsinalistarequnique.py
'''
Write a function solve that checks if all elements in a list are unique (no duplicates).

Example:
Input: [1, 2, 3, 4]
Output: True 
'''

def solve(arr):
  hashset = set()
  for i in range(len(arr)):
    if arr[i] in hashset:
      return False
    hashset.add(arr[i])
  return True

print(solve([1,2,3,4]))
print(solve([1,2,1,3]))

# Cool python one liner solution to this
def oneliner(arr):
  return len(arr) == len(set(arr))

print(oneliner([1,2,3,4]))
print(oneliner([1,2,1,3]))
